[5.7] Use functional mail button class names

sebastiandedeyne opened the pull request:
This PR replaces color names with functional labels for button classes. 

Today, if your application's primary color is something other than blue, you need to override the `button.blade.php` component if you don't want a `button-blue` class that isn't blue.

I stole Bootstrap's color names for the new button classes. If this gets accepted, I'll send a follow-up documentation PR.

sebastiandedeyne committed:
Use functional mail button class names

tillkruss commented:
Yeah, or send it to 5.7?

sebastiandedeyne commented:
@tillkruss This PR already targets 5.7

This would indeed break `color`. Keeping both defined is possible (then the CSS would target two selectors).

Another in-between option would be to keep calling it `color` but renaming the classes, so you'd have `'color' => 'primary'`. This would still break things, but less changes for the end user.

Up to you guys on how you want to handle this in the release! I'll update the PR with whatever you decide.

tillkruss commented:
Ah, right! I'd just have both class names.

taylorotwell commented:
Yeah I would just have both class names available in 5.7, and then we could probably list it as a deprecation and remove the old colors names in 5.8.

sebastiandedeyne committed:
Backwards compat

sebastiandedeyne commented:
Pushed some changes:

- Kept the `color` variable name, no more rename to `type`
- Changed `danger` to `error` to be in line with the notifications instead of Bootstrap colors
- Refactored the button in the notification view

taylorotwell commented:
Thanks, could you send a PR to `master` branch of docs?
